README.md

```diff
@@ -175,6 +175,221 @@ AVS Registry contracts have a stale view of operator shares in the delegation ma
 
+## Structure Documentation
+This AVS has three main participants:
+
+- Operator: The operator suscribes to NewTasks Events, and in case a new task is created, completes the task, calculates the response, signs it and sends it to the bls aggregation service.
+- Aggregator: The one who creates new tasks for the operators (through the on-chain Task Manager) every certain time. It also collects aggregated responses from the BLS aggregation service and sends them to the on-chain Task Manager, who then emits a TaskRespondedEvent.
+- Challenger: The Challenger subscribes to TaskRespondedEvents, and in case the response given by the aggregator differs from the challenger calculated response, it raises a challenge, that calls on-chain Task Manager, that verifies if the aggregator response was right. If it was not right, then the operator that signed the task will be slashed.
+
+Now we will focus on each to show how each one does each thing.
+
+### Operator
+
+The operator code can be found on [`/operator` folder](https://github.com/Layr-Labs/incredible-squaring-avs/tree/dev/operator).
+
+The operator main logic is focused on this segment from [operator.go](https://github.com/Layr-Labs/incredible-squaring-avs/blob/f8c379b151d8db778a12a5de1ba0266436d85366/operator/operator.go#L340-L363):
+
+```go
+for {
+	select {
+	case <-ctx.Done():
+		...
+	case err := <-metricsErrChan:
+		...
+	case err := <-sub.Err():
+		...
+	case newTaskCreatedLog := <-o.newTaskCreatedChan:
+		...
+	}
+}
+```
+
+The upper three cases are handling error cases, the fourth one is the one which pops from the channel subscribed to new task creation events, and handles the response logic:
+
+```go
+o.metrics.IncNumTasksReceived()
+taskResponse := o.ProcessNewTaskCreatedLog(newTaskCreatedLog)
+signedTaskResponse, err := o.SignTaskResponse(taskResponse)
+if err != nil {
+	continue
+}
+go o.aggregatorRpcClient.SendSignedTaskResponseToAggregator(signedTaskResponse)
+```
+
+The [`ProcessNewTaskCreatedLog` method](https://github.com/Layr-Labs/incredible-squaring-avs/blob/f8c379b151d8db778a12a5de1ba0266436d85366/operator/operator.go#L368-L394) generates the response to the new task:
+
+```go
+func (o *Operator) ProcessNewTaskCreatedLog(
+	newTaskCreatedLog *cstaskmanager.ContractIncredibleSquaringTaskManagerNewTaskCreated,
+) *cstaskmanager.IIncredibleSquaringTaskManagerTaskResponse {
+	...
+	numberSquared := big.NewInt(0).Exp(newTaskCreatedLog.Task.NumberToBeSquared, big.NewInt(2), nil)
+
+	...
+	taskResponse := &cstaskmanager.IIncredibleSquaringTaskManagerTaskResponse{
+		ReferenceTaskIndex: newTaskCreatedLog.TaskIndex,
+		NumberSquared:      numberSquared,
+	}
+	return taskResponse
+}
+```
+
+Here is the response calculation logic, and it would be the place to change if wanted to respond, for example, the cubed number instead. Note that the Response struct includes the number square because is parte of the Task Manager, that should be modified too.
```

Comment:
```suggestion
Here is the response calculation logic, and it would be the place to change if you wanted to compute, for example, the cubed number instead. Note that the `Response` struct includes the number square because it's part of the `TaskManager` contract bindings, which should be modified too.
```